http-server-netty/src/main/java/io/micronaut/http/server/netty/handler/accesslog/HttpAccessLogHandler.java

```diff
@@ -73,25 +87,42 @@ public HttpAccessLogHandler(String loggerName, String spec) {
      * @param spec The log format specification.
      */
     public HttpAccessLogHandler(Logger logger, String spec) {
+        this(logger, spec, null);
+    }
+
+    /**
+     * Creates a HttpAccessLogHandler.
+     *
+     * @param logger A logger. Will log at info level.
+     * @param spec The log format specification.
+     * @param uriInclusion A filtering Predicate that will be checked per URI.
+     */
+    public HttpAccessLogHandler(Logger logger, String spec, Predicate<String> uriInclusion) {
         super();
         this.logger = logger == null ? LoggerFactory.getLogger(HTTP_ACCESS_LOGGER) : logger;
         this.accessLogFormatParser = new AccessLogFormatParser(spec);
+        this.uriInclusion = uriInclusion;
     }
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Http2Exception {
         if (logger.isInfoEnabled() && msg instanceof HttpRequest) {
             final SocketChannel channel = (SocketChannel) ctx.channel();
             final HttpRequest request = (HttpRequest) msg;
-            final HttpHeaders headers = request.headers();
-            // Trying to detect http/2
-            String protocol;
-            if (headers.contains(ExtensionHeaderNames.STREAM_ID.text()) || headers.contains(ExtensionHeaderNames.SCHEME.text())) {
-                protocol = H2_PROTOCOL_NAME;
+            AccessLog accessLog = accessLog(channel);
+            if (uriInclusion == null || uriInclusion.test(request.uri())) {
+                final HttpHeaders headers = request.headers();
+                // Trying to detect http/2
+                String protocol;
+                if (headers.contains(ExtensionHeaderNames.STREAM_ID.text()) || headers.contains(ExtensionHeaderNames.SCHEME.text())) {
+                    protocol = H2_PROTOCOL_NAME;
+                } else {
+                    protocol = request.protocolVersion().text();
+                }
+                accessLog.onRequestHeaders(channel, request.method().name(), request.headers(), request.uri(), protocol);
             } else {
-                protocol = request.protocolVersion().text();
+                accessLog.exclude();
```

Comment:
@graemerocher Would it be safe here to remove the handler from the pipeline? Setting an excluded flag and checking it is extra overhead

Reply:
Nice idea!

```
                ctx.pipeline().remove(this);
```

Seems to work in the test...  Checking, and if there's no issues I can find I'll push

Reply:
Load tested it with a filtered url and a non-filtered url

```
ab -k -n 100000 -r -c 5 http://localhost:8080/path/one & ab -k -r -n 100000 -c 5 http://localhost:8080/something
```

And we didn't get any errors.  I'm not a Netty-guru (yet) though, so @graemerocher may know if this is a dangerous thing to be doing

Anyway, pushed the change 👍

Reply:
Right, so that doesn't work with keep-alive connections (all logs get dropped after the first excluded one removes the handler from the pipeline)

Reverted back to the field